src/Analyser/MutatingScope.php

```diff
@@ -1236,7 +1242,7 @@ private function resolveType(string $exprString, Expr $node): Type
 						new VoidType(),
 					]);
 				} else {
-					$returnType = $arrowScope->getType($node->expr);
+					$returnType = $arrowScope->getKeepVoidType($node->expr);
```

Comment:
does it make sense to handle a "void" return type of a arrow function? can arrow functions have a void return type?

Reply:
Try to change it and see what fails. I think it's valid, at least for a throw expr or calling never function.

Reply:
just found https://github.com/phpstan/phpstan-src/pull/2778#issuecomment-1833752355 which answers this question, I guess

Reply:
hmm locally I did not get a test failure after changing it. lets see https://github.com/phpstan/phpstan-src/pull/2873